src/main/java/com/limechain/trie/structure/slab/Slab.java

```diff
@@ -16,70 +15,157 @@
 import java.util.stream.IntStream;
 import java.util.stream.Stream;
 
+/**
+ * Represents a slab data structure for efficient management of a fixed type of objects.
+ * It provides methods for adding, removing, and accessing elements, along with utilities
+ * for managing the slab's capacity and contents.
+ *
+ * @param <T> the type of elements held in this slab
+ */
 public class Slab<T> implements Iterable<Pair<Integer, T>> {
     private ArrayList<T> storage;
     private int newIndex;
     private Queue<Integer> freeIndices;
+    private int size;
 
+    /**
+     * Constructs an empty Slab with default initial capacity of 0.
```

Comment:
Maybe the default capacity should be sth like `10` or `13`, similar to that Java's devs have decided for `ArrayList`?

Reply:
I've set it to 10